### DISCUSSION TITLE
reveal-js: wrap code does not work

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#11157] [Q&A] [answered]
### Bug description

Hi,
I built sample slides with some html code to display. I applied wrap, but no carriage return is applied.

The rendering https://aborruso.quarto.pub/lorem-ipsum

### Steps to reproduce

This is the sample code.

I have published it using `quarto publish sample.qmd`

````md
---
title: "Lorem Ipsum"
author: "John Doe"
format:
  revealjs:
    incremental: true
    lang: it-IT
    code-overflow: wrap
---

## HTML no wrap 1

```{.html}
<span class="title-file" data-size="326.0 kb" data-extension="csv" data-url="/content/dam/sogei-ng/open-data-b/OpenData_PNRR_map_indicatori_comuni.csv">
    OpenData_PNRR_map_indicatori_comuni.csv -
    <span class="extension">(326.0 kb)</span>
</span>
<span class="title-file hidden" data-size="682.0 kb" data-extension="json" data-url="/content/dam/sogei-ng/open-data-b/OpenData_PNRR_map_indicatori_comuni_v7.json">
    OpenData_PNRR_map_indicatori_comuni_v7.json -
    <span class="extension">(682.0 kb)</span>
</span>
```

## HTML no wrap 2

```{.html .code-overflow-wrap}
<span class="title-file" data-size="326.0 kb" data-extension="csv" data-url="/content/dam/sogei-ng/open-data-b/OpenData_PNRR_map_indicatori_comuni.csv">
    OpenData_PNRR_map_indicatori_comuni.csv -
    <span class="extension">(326.0 kb)</span>
</span>
<span class="title-file hidden" data-size="682.0 kb" data-extension="json" data-url="/content/dam/sogei-ng/open-data-b/OpenData_PNRR_map_indicatori_comuni_v7.json">
    OpenData_PNRR_map_indicatori_comuni_v7.json -
    <span class="extension">(682.0 kb)</span>
</span>
```

````

### Expected behavior

Having the carriage return and having all the text readable without doing horizontal scroll

### Actual behavior

I have no wrap, either using `code-overflow: wrap` or using `.code-overflow-wrap`

![Image](https://github.com/user-attachments/assets/2e46745d-fa84-4bca-a084-6616e0d31391)

### Your environment

- IDE: vscode
- OS: debian 12.5

### Quarto check output

```
Quarto 1.5.56
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.56
      Path: /home/aborruso/bin/quarto-1.5.56/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /usr/bin
      Version: 2022

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.2
      Path: /usr/bin/python3
      Jupyter: 5.3.2
      Kernels: bash, python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.2.2
      Path: /usr/lib/R
      LibPaths:
        - /usr/local/lib/R/site-library
        - /usr/lib/R/site-library
        - /usr/lib/R/library
      knitr: 1.45
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK
```

**Comment by cscheid** [comment]
(When you have future questions, could you consider opening a discussion before an issue? It helps us organize the forums a bit better. Thanks!)

~~I think what's going on here is that our wrapping style only kicks in for elements that are generated as code output from a code cell.~~

Ok. Our docs do promise support for it, [but only in HTML](https://quarto.org/docs/output-formats/html-code.html#code-overflow).

**Reply by cscheid** [reply]
In `html`, this works:

<img width="1126" alt="image" src="https://github.com/user-attachments/assets/20bf26bd-3bca-4447-99bc-32f7b3d1f82d">

In `revealjs`, we don't promise support for it. Ideally we'd support that, but it's not a bug.

**Reply by cscheid** [reply]
Tracking here: https://github.com/quarto-dev/quarto-cli/issues/11158

**Reply by aborruso** [reply]
First of all, thank you

> When you have future questions, could you consider opening a discussion before an issue? It helps us organize the forums a bit better. Thanks!)

Before opening an issue, I verified that it was an option for reveal: https://quarto.org/docs/reference/formats/presentations/revealjs.html#code

I'm a bad person for so many reasons, but if I see it in the documentation and it doesn't work, it looks like a bug to me. Is that reasoning wrong?

![image](https://github.com/user-attachments/assets/ecca5e0a-8e05-4a73-b81f-08ff292b60df)

**Reply by mcanouil** [reply]
FYI, the reference pages are automatically generated but set manually in the following:

- https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/schema/cell-codeoutput.yml#L63-L77

This issue could either be a bug, where something documented does not work, or a documentation oversight, where something not supported is listed.
Either way, thank you for your report. 

We apologise for the confusion caused. It appears that users frequently use options across formats, regardless of the provided documentation. This assumption was incorrect, and for that, we are sorry.

In the future, please feel free to link directly to the relevant documentation. This will help make things clearer for everyone involved.

**Reply by cscheid** [reply, marked as answer]
I'll reiterate exactly what @mcanouil said; this was _my_ confusion - I apologize!

I personally got confused by the fact that we describe `code-overflow` in our `html` page  https://quarto.org/docs/output-formats/html-code.html#code-overflow but not in our revealjs guide. But if the revealjs reference page includes a link to it, then it's definitely a bug.

**Reply by aborruso** [reply]
Once again, thank you. What you and so many others do for this project is extraordinary; “quarto” is a great product, and it is so because of you.
I feel the need to add some notes.

Very recently [another issue of mine](https://github.com/quarto-dev/quarto-cli/issues/10710), which was a bug, produced a discussion that was taking it elsewhere. But it was well described, it was very much in focus.
And even then it shifted in some way to the fact that I was not performing my user role well.

I love “quarto,” even if I can't make better contributions than a few bugs discovered, and creating a few sites that get quarto discovered.
We users certainly make a lot of mistakes, and we are not that important to “quarto.” But we also often put in attention and love for the project.

@cscheid a question: will you now move this discussion back to issues?

Thank you and have a nice day

**Reply by aborruso** [reply]
> @cscheid a question: will you now move this discussion back to issues?

I see it, it's #11158
